File: msg-entity/src/test/java/dev/number6/entity/adaptor/FakeComprehensionAdaptor.java
package dev.number6.entity.adaptor;

import dev.number6.comprehend.port.ComprehensionPort;
import dev.number6.comprehend.results.PresentableEntityResults;
import dev.number6.comprehend.results.PresentableKeyPhrasesResults;
import dev.number6.comprehend.results.PresentableSentimentResults;
import dev.number6.message.ChannelMessages;

public class FakeComprehensionAdaptor implements ComprehensionPort {

    @Override
    public PresentableEntityResults getEntitiesForSlackMessages(ChannelMessages channelMessages) {
        throw new UnsupportedOperationException("cannot get entities in Fake Comprehension Adaptor");
    }

    @Override
    public PresentableSentimentResults getSentimentForSlackMessages(ChannelMessages channelMessages) {
        throw new UnsupportedOperationException("cannot get sentiment in Fake Comprehension Adaptor");
    }

    @Override
    public PresentableKeyPhrasesResults getKeyPhrasesForSlackMessages(ChannelMessages channelMessages) {
        throw new UnsupportedOperationException("cannot get key phrases in Fake Comprehension Adaptor");
    }
}

File: msg-key-phrases/src/test/java/dev/number6/keyphrases/adaptor/FakeComprehensionAdaptor.kt
package dev.number6.keyphrases.adaptor

import dev.number6.comprehend.port.ComprehensionPort
import dev.number6.comprehend.results.PresentableEntityResults
import dev.number6.comprehend.results.PresentableKeyPhrasesResults
import dev.number6.comprehend.results.PresentableSentimentResults
import dev.number6.message.ChannelMessages

class FakeComprehensionAdaptor : ComprehensionPort {
    override fun getEntitiesForSlackMessages(channelMessages: ChannelMessages): PresentableEntityResults {
        throw UnsupportedOperationException("cannot get entities in Fake Comprehension Adaptor")
    }

    override fun getSentimentForSlackMessages(channelMessages: ChannelMessages): PresentableSentimentResults {
        throw UnsupportedOperationException("cannot get sentiment in Fake Comprehension Adaptor")
    }

    override fun getKeyPhrasesForSlackMessages(channelMessages: ChannelMessages): PresentableKeyPhrasesResults {
        throw UnsupportedOperationException("cannot get key phrases in Fake Comprehension Adaptor")
    }
}